pkg/libvmi/vmi.go

```diff
@@ -84,6 +84,12 @@ func WithNamespace(namespace string) Option {
 	}
 }
 
+func WithName(name string) Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		vmi.Name = name
+	}
+}
```

Comment:
why do you need this change ? I can't reason why/how a VM can avoid having a name.

Couldn't you just run with the [random name the vm is assigned](https://github.com/kubevirt/kubevirt/blob/3e62475af965aa9799b1880205cd3e75c2425e33/pkg/libvmi/vmi.go#L39) ?

Reply:
We need explicit name and not random, to easily have expected ownerRef / IPAMClaim name etc
I think it makes the test nicer with it (it was already the state few commits ago, just due to refactors of using libvmi had to add it)

Reply:
> I don't see a reason to generate random and then get it to compare to it

a left-over from a previous test will pretty much break the next one. 

Using generated names is usually good practice. This goes in the other direction. I will not block on this, I just think it is preferable to create a vm with a random name, then get whatever is returned from create. You get that for free.

Reply:
on it